gmime/meson.build

```diff
@@ -0,0 +1,221 @@
+  'gmime-parse-utils.c',
+  'gmime-part.c',
+  'gmime-part-iter.c',
+  'gmime-pkcs7-context.c',
+  'gmime-references.c',
+  'gmime-signature.c',
+  'gmime-stream-buffer.c',
+  'gmime-stream.c',
+  'gmime-stream-cat.c',
+  'gmime-stream-file.c',
+  'gmime-stream-filter.c',
+  'gmime-stream-fs.c',
+  'gmime-stream-gio.c',
+  'gmime-stream-mem.c',
+  'gmime-stream-mmap.c',
+  'gmime-stream-null.c',
+  'gmime-stream-pipe.c',
+  'gmime-text-part.c',
+  'gmime-utils.c',
+  'internet-address.c',
+)
+
+private = [
+  'gmime-charset-map-private.h',
+  'gmime-table-private.h',
+  'gmime-parse-utils.h',
+  'gmime-gpgme-utils.h',
+  'gmime-internal.h',
+  'gmime-common.h',
+  'gmime-events.h',
+]
+
+libgmime_headers = [
+  'gmime-application-pkcs7-mime.h',
+  'gmime-autocrypt.h',
+  'gmime-certificate.h',
+  'gmime-charset.h',
+  'gmime-content-type.h',
+  'gmime-crypto-context.h',
+  'gmime-data-wrapper.h',
+  'gmime-disposition.h',
+  'gmime-encodings.h',
+  'gmime-error.h',
+  'gmime-filter-basic.h',
+  'gmime-filter-best.h',
+  'gmime-filter-charset.h',
+  'gmime-filter-checksum.h',
+  'gmime-filter-dos2unix.h',
+  'gmime-filter-enriched.h',
+  'gmime-filter-from.h',
+  'gmime-filter-gzip.h',
+  'gmime-filter.h',
+  'gmime-filter-html.h',
+  'gmime-filter-openpgp.h',
+  'gmime-filter-smtp-data.h',
+  'gmime-filter-strip.h',
+  'gmime-filter-unix2dos.h',
+  'gmime-filter-windows.h',
+  'gmime-filter-yenc.h',
+  'gmime-format-options.h',
+  'gmime-gpg-context.h',
+  'gmime.h',
+  'gmime-header.h',
+  'gmime-iconv.h',
+  'gmime-iconv-utils.h',
+  'gmime-message.h',
+  'gmime-message-part.h',
+  'gmime-message-partial.h',
+  'gmime-multipart-encrypted.h',
+  'gmime-multipart.h',
+  'gmime-multipart-signed.h',
+  'gmime-object.h',
+  'gmime-param.h',
+  'gmime-parser.h',
+  'gmime-parser-options.h',
+  'gmime-part.h',
+  'gmime-part-iter.h',
+  'gmime-pkcs7-context.h',
+  'gmime-references.h',
+  'gmime-signature.h',
+  'gmime-stream-buffer.h',
+  'gmime-stream-cat.h',
+  'gmime-stream-file.h',
+  'gmime-stream-filter.h',
+  'gmime-stream-fs.h',
+  'gmime-stream-gio.h',
+  'gmime-stream.h',
+  'gmime-stream-mem.h',
+  'gmime-stream-mmap.h',
+  'gmime-stream-null.h',
+  'gmime-stream-pipe.h',
+  'gmime-text-part.h',
+  'gmime-utils.h',
+  'gmime-version.h',
+  'internet-address.h',
+]
+
+inc_conf = include_directories('..', '../util')
+
+version_data = configuration_data()
+version_data.set('GMIME_MAJOR_VERSION', GMIME_MAJOR_VERSION)
+version_data.set('GMIME_MINOR_VERSION', GMIME_MINOR_VERSION)
+version_data.set('GMIME_MICRO_VERSION', GMIME_MICRO_VERSION)
+version_data.set('GMIME_BINARY_AGE', GMIME_BINARY_AGE)
+version_data.set('GMIME_INTERFACE_AGE', GMIME_INTERFACE_AGE)
+configure_file(input: 'gmime-version_meson.h.in',
+  output: 'gmime-version.h',
+  configuration : version_data
+)
+
+gen_table = executable('gen-table', 'gen-table.c')
+charset_map = executable('charset-map', 'charset-map.c', dependencies : deps, 
+  include_directories: inc_conf)
```

Comment:
Since cross compilation came up... you can mark an executable as `native: true` to make it be compiled with the build machine compiler instead of the cross-compiler, although if you need an exe_wrapper anyway that might not matter so much.

Reply:
fixed